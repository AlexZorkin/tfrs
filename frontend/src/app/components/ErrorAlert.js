import React from 'react';
import PropTypes from 'prop-types';

const ErrorAlert = props => (
  <div className="alert alert-danger error-alert" role="alert">
    <strong> {props.title} </strong>
    <div>{props.message}</div>
  </div>
);

ErrorAlert.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

export default ErrorAlert;
