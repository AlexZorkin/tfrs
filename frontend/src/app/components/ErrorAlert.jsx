import React, {Component} from 'react';

export default class ErrorAlert extends Component {
  render () {
    return (
      <div>
        <h1>{this.props.title}</h1>
        <p>{this.props.message}</p>
      </div>
    );
  }
}
