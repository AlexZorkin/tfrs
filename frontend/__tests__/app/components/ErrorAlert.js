import React from 'react';
import renderer from 'react-test-renderer';

import ErrorAlert from '../../../src/app/components/ErrorAlert';

test('ErrorAlert should display the title and message as expected', () => {
  const component = renderer.create(<ErrorAlert title="Title" message="Message" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
