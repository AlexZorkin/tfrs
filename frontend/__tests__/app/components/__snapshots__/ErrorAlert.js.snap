// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert should display the title and message as expected 1`] = `
<div>
  <h1>
    Title
  </h1>
  <p>
    Message
  </p>
</div>
`;
